torchdynamo/optimizations/distributed.py

```diff
@@ -19,15 +22,35 @@ def args_str(args):
         return str(args)
 
 
+@dataclass
+class Bucket:
+    size: int = 0
+    params: List[str] = field(default_factory=list)
+    nodes: List[fx.Node] = field(default_factory=list)
+
+
+def pretty_print_buckets(buckets: List[Bucket]):
+    headers = ("Index", "Size (b)", "Param Names")
+    rows = []
+    for idx, bucket in enumerate(reversed(buckets)):
+        rows.append((idx, bucket.size, bucket.params[0]))
+        for param in bucket.params[1:]:
+            rows.append((None, None, param))
+    print(tabulate(rows, headers=headers, tablefmt="simple_grid"))
+
+
 class DDPOptimizer:
     def __init__(
         self,
         bucket_bytes_cap: int,
         parameters_to_ignore: List[str],
         backend_compile_fn,
         debug=False,
+        # TODO(whc) align this with DDP, a special case to get one small bucket started early
```

Comment:
@mrshenli you told me about this a while ago. What would be the best way to get the number?  is there a parameter/api on DDP object i can query, or should i hardcode for now?

Reply:
probably `dist._DEFAULT_FIRST_BUCKET_BYTES` ? https://github.com/pytorch/pytorch/blob/master/torch/nn/parallel/distributed.py#L748-L750